### PR TITLE
Add initial OWIN support through a separate Datadog.Trace.Owin library

### PR DESCRIPTION
**Pull request by zacharycmontoya**
OWIN allows .NET Framework web applications to be decoupled from the IIS web server. As a precursor to the ASP.NET Core pipeline, OWIN allows for configuration through middleware. This PR adds a middleware component that will trace the rest of the request processing pipeline. The middleware should be added at the beginning of the pipeline for most visibility, and it can be added through one of two mechanisms:
1. `IAppBuilder.UseDatadogTracingOwinMiddleware()` extension method
1. `IAppBuilder.Use<DatadogTracingOwinMiddleware>()`

All changes include:
- Modify `StartDistributedTrace` tool to allow URL to be set by console input
- Modify `SpanContextPropagator` to take delegates for extracting context and header tags instead of requiring new classes (feedback welcome, this PR can be completed through original mechanism though I'm a fan of this new method)
- Add test application `Samples.Owin.WebApi` to integration tests

### CODE REVIEW

File: src/Datadog.Trace.Owin/OwinTags.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    internal class OwinTags : WebTags
+    {
+        private const string ComponentName = "Owin";
+
+        private static readonly IProperty<string>[] AspNetCoreTagsProperties =
+            WebTagsProperties.Concat(
+                new ReadOnlyProperty<OwinTags, string>(Trace.Tags.InstrumentationName, t => t.InstrumentationName));
+
+        public string InstrumentationName => ComponentName;
```
Comment (lucaspimentel) [inline src/Datadog.Trace.Owin/OwinTags.cs:15]: Not necessary for in this PR, but we should probably promote "component" to a top-level tag at some point and make sure it's included in all integrations.

File: src/Datadog.Trace/SpanContextPropagator.cs [hunk 2]
```diff
@@ -60,6 +60,39 @@ public void Inject<T>(SpanContext context, T headers)
                 samplingPriority?.ToString(InvariantCulture));
         }
 
+        /// <summary>
+        /// Propagates the specified context by adding new headers to a <see cref="IHeadersCollection"/>.
```
Comment (lucaspimentel) [inline src/Datadog.Trace/SpanContextPropagator.cs:64]: This comment shouldn't mention `IHeadersCollection`.

**Comment by lucaspimentel** [comment]
@zacharycmontoya, what's the status of this PR? Do we still want to review and merge?

**Comment by zacharycmontoya** [comment]
Given the other work going and lack of urgency, there's no need to merge this now. I'll close this and keep the branch around for later in case we decide to pick up the work again.

**Pull request closed without merging**